src/ixdat/readers/cinfdata_db.py

```diff
@@ -0,0 +1,438 @@
+                column_name = self.group_meta[key]["mass_label"]
+                if self.verbose:
+                    print("Col name: ", column_name)
+                # unixtime = self.group_meta[key]["unixtime"]
+                # tstamp = float(unixtime)
+
+                tcol = self.group_data[key][:, 0]
+                vcol = self.group_data[key][:, 1]
+
+                tseries = TimeSeries(
+                    name=column_name + "-x",
+                    unit_name=get_column_unit(column_name + "-x") or "s",
+                    data=tcol,
+                    tstamp=self.tstamp,
+                )
+
+                vseries = ValueSeries(
+                    name=column_name,
+                    data=vcol,
+                    tseries=tseries,
+                    unit_name=get_column_unit(column_name + "-y"),
+                )
+                data_series_list.append(tseries)
+                data_series_list.append(vseries)
+            else:
+                pass
+
+        obj_as_dict = dict(
+            name=self.name,
+            sample_name=self.sample_name,
+            technique=self.technique,
+            reader=self,
+            series_list=data_series_list,
+            tstamp=self.tstamp,
+        )
+
+        if not data_series_list: #obj_as_dict:
+            warnings.warn(
+                f"No mass spec data was found using '{self.token}' "
+                f" and group_column: '{self.grouping_column}'",
+                stacklevel=2,
+            )
+            return None
+
+        return obj_as_dict
+
+    def read_spectrums(self, **kwargs):
+        """
+        Download spectrums from cinfdata_database with either a timestamp or a comment.
+        When all associated spectras have been downloaded and added to self.spectrum_list,
+        this method will try to:
+            1. figure out if any spectras were able to be downloaded; if none, return None
+            2. figure out if this method was used to add mass_scans to another measurement;
+               return list of Spectrum to be added to that measurement
+            3. figure out if one and only one spectrum was downloaded; return that single Spectrum
+               as the first spectrum in self.spectrum_list[0]
+            4. figure out if all spectras downloaded are associated enough to be a SpectrumSeries
+               (They have to be equal in dimensions like mass scans) (multiple mass scans over time
+               versus XPS spectra from different regions); return SpectrumSeries
+            5. If not one and only one Spectrum was read and the multiple of spectras read were not
+               possible to combine in a SpectrumSeries, return self.spectrum_list
+        """
+        db = plugins.cinfdata.connect(setup_name=self.setup_name, grouping_column=self.grouping_column)
+        self.group_data = db.get_data_group(self.token)
+        self.group_meta = db.get_metadata_group(self.token)
```

Comment:
I don't understand the line below. It looks like all the meta data is in the first key of the `group_meta` dictionary? That doesn't make sense to me... dictionaries are in principle not ordered, and the data is usually in the values (and just pointed to by the keys).... 
The line probably makes perfect sense in the context, but a comment explaining the structure of `group_meta` would help!

Reply:
I have left a comment about that any key ( unique measurement ) holds the necessary information of metadata to set tstamp, sample_name and name of measurement. 

So yes, it is picked at random because metadata associated with any key could give the information. 
I think the readability is higher / easier when this specific metadata is extracted once.

But maybe the line should be placed down together with the function `set_sample_name(), set_name(), set_tstamp()`

Reply:
What is the type of the key and the type of the value of the dictionary `group_metadata`? Are you saying that the key is an ixdat `Measurement` object? That doesn't make any sense to me, as it is a method defined in the cinfdata module, which doesn't know about ixdat.
We just need a comment explaining what is mapped to what in this dictionary!

Reply:
On the other hand, there was no need to move setting the sample name, name, and tstamp to other methods. (Methods without arguments or returns that are only called fom one place have no point, I think.) I think it's an overuse of the reader object, and makes the code harder to read.